sv_pipeline/genome/load_data_tests.py

```diff
@@ -0,0 +1,279 @@
+            mock.call('Subsetting to 4 samples (remapping 1 samples)'),
+        ]
+        mock_logger.info.assert_has_calls(calls)
+        mock_get_sample.assert_called_with('test_guid', 'WGS')
+        mock_get_remap.assert_called_with('test_guid', 'WGS')
+        self.assertEqual(rows.count(), 5)
+        self.assertEqual(set(rows.genotypes.map(lambda x: x.sample_id).take(1)[0]), {'SAMPLE-1-REMAP', 'SAMPLE-3', 'SAMPLE-5'})
+
+        mock_logger.reset_mock()
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=True)
+        self.assertEqual(rows.count(), 5)
+        self.assertEqual(set(rows.genotypes.map(lambda x: x.sample_id).take(1)[0]), {'SAMPLE-1', 'SAMPLE-2', 'SAMPLE-3', 'SAMPLE-4', 'SAMPLE-5'})
+
+    @mock.patch('sv_pipeline.genome.load_data.load_gencode')
+    def test_annotation(self, mock_load_gencode):
+        mock_load_gencode.return_value = GENE_ID_MAPPING
+        rows = subset_mt('test_guid', self.mt, skip_sample_subset=True)
+        rows = annotate_fields(rows, TEST_GENCODE_RELEASE, TEST_GENCODE_PATH)
+        mock_load_gencode.assert_called_with(TEST_GENCODE_RELEASE, download_path=TEST_GENCODE_PATH)
+        row_list = rows.take(5)
+        self.assertListEqual([row_list[0], row_list[2], row_list[4]], hl.eval([VARIANT0, VARIANT2, VARIANT4]))
+
+    @mock.patch('sv_pipeline.genome.load_data.os')
+    @mock.patch('sv_pipeline.genome.load_data.ElasticsearchClient')
+    @mock.patch('sv_pipeline.genome.load_data.get_es_index_name')
+    def test_export_to_es(self, mock_get_index, mock_es_client, mock_os):
+        mock_get_index.return_value = TEST_INDEX_NAME
+        mock_os.environ.get.return_value = TEST_PASSWORD
+        mock_es = mock_es_client.return_value
+        rows = self.mt.rows().head(5)
+        export_to_es(rows, TEST_INPUT_DATASET, TEST_GUID, TEST_HOST, TEST_PORT, TEST_BLOCK_SIZE, TEST_NUM_SHARDS)
+        mock_get_index.assert_called_with(TEST_GUID, {'genomeVersion': '38', 'sampleType': WGS_SAMPLE_TYPE,
+                                                      'datasetType': 'SV', 'sourceFilePath': TEST_INPUT_DATASET})
+        mock_os.environ.get.assert_called_with('PIPELINE_ES_PASSWORD', '')
+        mock_es_client.assert_called_with(host=TEST_HOST, port=TEST_PORT, es_password=TEST_PASSWORD)
+        mock_es.export_table_to_elasticsearch.assert_called_with(
+            mock.ANY,
+            index_name=TEST_INDEX_NAME,
+            block_size=TEST_BLOCK_SIZE,
+            num_shards=TEST_NUM_SHARDS,
+            delete_index_before_exporting=True,
+            export_globals_to_index_meta=True,
+            verbose=True,
+        )
+        self.assertEqual(hl.eval(mock_es.export_table_to_elasticsearch.call_args.args[0].globals),
+                         hl.eval(hl.struct(genomeVersion='38', sampleType=WGS_SAMPLE_TYPE,
+                                           datasetType='SV', sourceFilePath=TEST_INPUT_DATASET)))
+
+    @mock.patch('sv_pipeline.genome.load_data.hl')
```

Comment:
why do you mock hail in this test but none of the other tests?

Reply:
For `main()`, it calls `hl.init()` which can only be called for once. In the test setup, `hl.init()` has been called. Therefore, `hl` has to be mocked for testing `main()`. I think it is better to mock all `hl` because we don't want to test third-party libraries ' behaviors. But, for `subset_mt()` and `annotate_fields()`, it is hard to test if mock `hl`. So, I leave it as is.

Reply:
okay, but then you should assert that the mocked init is called